### ISSUE TITLE
Bug with extracting helical particles from eman2 box files

### ISSUE THREAD

**Issue by JoeBeton** [3dem/relion#580]
Hello, I hope this is the correct place to report this issue. 

I have encountered a problem when trying to extract helical particles from .box files generated by crYOLO. My extract jobs consistently generated an error stating: 

> helix.cpp::convertEmanHelicalSegmentCoordsToMetaDataTable(): Wrong dimensions or box size!

which occurred regardless of which box size I used (full error message shown below). I am sure the box files are formatted correctly and the coordinates are pointing to the particles. I managed to fix the problem by changing line 602 in preprocessing.cpp from:

```C++
convertEmanHelicalSegmentCoordsToMetaDataTable(fn_coord, MD, total_segments, total_tubes, xdim, ydim, extract_size, helical_bimodal_angular_priors);
```
to (including angpix when calling the function):

```C++
convertEmanHelicalSegmentCoordsToMetaDataTable(fn_coord, MD, total_segments, total_tubes, angpix, xdim, ydim, extract_size, helical_bimodal_angular_priors);
```
After this change the error no longer occurred, the job ran to completion and the particles looked correct. Hope this is helpful!

Thanks, 

Joe

**Environment:**
 - OS: CentOS 6.1
 - MPI runtime: 
 - RELION-3.1 commit f04522
 - Memory: 256 GB
 - GPU: none used

**Dataset:**
 - Box size: 128 pix (binned from 416)
 - Pixel size: 3.575 A (binned from 1.1 A)
 - Number of particles: 2,338,810
 - Description: Helical filament

**Job options:**
 - Type of job: Particle extraction
 - Number of MPI processes: 20
 - Number of threads: 1

 - Full command:
```
which relion_preprocess --i Select/job011/micrographs.star --coord_dir Import/job023/ --coord_suffix .box --part_star Extract/job044/particles.star --part_dir Extract/job044/ --extract --extract_size 416 --scale 128 --norm --bg_radius 0 --white_dust 3 --black_dust 3 --invert_contrast  --helix --helical_outer_diameter 270 --helical_bimodal_angular_priors --only_do_unfinished   --pipeline_control Extract/job044/
```

**Error message:**
```
in: /s/emib/s/relion/v3.1/relion/src/helix.cpp, line 2665
ERROR:
helix.cpp::convertEmanHelicalSegmentCoordsToMetaDataTable(): Wrong dimensions or box size!
=== Backtrace  ===
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(_ZN11RelionErrorC1ERKSsS1_l+0x41) [0x44fc21]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(_Z46convertEmanHelicalSegmentCoordsToMetaDataTableR8FileNameR13MetaDataTableRiS3_ddddb+0x101e) [0x4fc32e]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(_ZN13Preprocessing22readHelicalCoordinatesE8FileNameS0_R13MetaDataTable+0x436) [0x45fe46]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(_ZN13Preprocessing31extractParticlesFromFieldOfViewE8FileNamel+0x8d5) [0x4665d5]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(_ZN16PreprocessingMpi19runExtractParticlesEv+0x3c6) [0x436586]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(_ZN16PreprocessingMpi3runEv+0x12) [0x4369f2]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi(main+0x3b) [0x42abcb]
/lib64/libc.so.6(__libc_start_main+0x100) [0x337c61ed20]
/s/emib/s/relion/v3.1/beta2/bin/relion_preprocess_mpi() [0x42b575]
```

**Comment by biochem-fan** [comment]
Thank you very much for report. @scheres fixed this in RELION 3.1. We will upload it soon.

**Comment by biochem-fan** [comment]
Repaired in fa89642.
